mne_bids/utils.py

```diff
@@ -0,0 +1,212 @@
+
+
+def make_bids_filename(subject=None, session=None, task=None,
+                       acquisition=None, run=None, processing=None,
+                       recording=None, suffix=None, prefix=None):
+    """Create a BIDS filename from its component parts.
+
+    BIDS filename prefixes have one or more pieces of metadata in them. They
+    must follow a particular order, which is followed by this function. This
+    will generate the *prefix* for a BIDS file name that can be used with many
+    subsequent files, or you may also give a suffix that will then complete
+    the file name.
+
+    Note that all parameters are not applicable to each kind of data. For
+    example, electrode location TSV files do not need a task field.
+
+    Parameters
+    ----------
+    subject : str | None
+        The subject ID. Corresponds to "sub".
+    session : str | None
+        The session for a item. Corresponds to "ses".
+    task : str | None
+        The task for a item. Corresponds to "task".
+    acquisition: str | None
+        The acquisition parameters for the item. Corresponds to "acq".
+    run : int | None
+        The run number for this item. Corresponds to "run".
+    processing : str | None
+        The processing label for this item. Corresponds to "proc".
+    recording : str | None
+        The recording name for this item. Corresponds to "recording".
+    suffix : str | None
+        The suffix of a file that begins with this prefix. E.g., 'audio.wav'.
+    prefix : str | None
+        The prefix for the filename to be created. E.g., a path to the folder
+        in which you wish to create a file with this name.
+
+    Returns
+    -------
+    filename : str
+        The BIDS filename you wish to create.
+
+    Examples
+    --------
+    >>> print(make_bids_filename(subject='test', session='two', task='mytask', suffix='data.csv')) # noqa
+    sub-test_ses-two_task-mytask_data.csv
+    """
+    order = OrderedDict([('sub', subject),
+                         ('ses', session),
+                         ('task', task),
+                         ('acq', acquisition),
+                         ('run', run),
+                         ('proc', processing),
+                         ('recording', recording)])
+    if order['run'] is not None and not isinstance(order['run'], str):
+        # Ensure that run is a string
+        order['run'] = '{:02}'.format(order['run'])
+
+    _check_types(order.values())
+
+    if not any(isinstance(ii, str) for ii in order.keys()):
+        raise ValueError("At least one parameter must be given.")
+    filename = []
+    for key, val in order.items():
+        if val is not None:
+            filename.append('%s-%s' % (key, val))
+    if isinstance(suffix, str):
+        filename.append(suffix)
+    filename = '_'.join(filename)
+    if isinstance(prefix, str):
+        filename = os.path.join(prefix, filename)
+    return filename
+
+
+def make_bids_folders(subject, session=None, kind=None, root=None,
```

Comment:
should we include an overwrite in case the some or all the folders exist?

Reply:
+1